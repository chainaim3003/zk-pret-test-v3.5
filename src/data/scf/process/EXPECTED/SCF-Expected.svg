<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1600 700">
  <!-- Background -->
  <rect width="1600" height="700" fill="#f8f9fa"/>
  
  <!-- Title -->
  <text x="800" y="30" text-anchor="middle" font-family="Arial, sans-serif" font-size="18" font-weight="bold" fill="#2563eb">
    SCF: Supply Chain Finance Process (Expected Pattern)
  </text>
  <text x="800" y="50" text-anchor="middle" font-family="Arial, sans-serif" font-size="12" fill="#64748b">
    Pattern: a(b|c)(c|b)d(e|f)fg (4 possible paths)
  </text>
  
  <!-- Start Event -->
  <circle cx="80" cy="250" r="18" fill="#10b981" stroke="#059669" stroke-width="2"/>
  <text x="80" y="255" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" font-weight="bold" fill="white">S</text>
  <text x="80" y="280" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" fill="#374151">Start</text>
  
  <!-- Node 1: First Event -->
  <circle cx="180" cy="250" r="15" fill="#3b82f6" stroke="#2563eb" stroke-width="2"/>
  <text x="180" y="255" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" font-weight="bold" fill="white">1</text>
  <text x="180" y="280" text-anchor="middle" font-family="Arial, sans-serif" font-size="9" fill="#374151">Process</text>
  
  <!-- FIRST BRANCH: (b|c) -->
  <circle cx="300" cy="190" r="15" fill="#f59e0b" stroke="#d97706" stroke-width="2"/>
  <text x="300" y="195" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" font-weight="bold" fill="white">2</text>
  <text x="300" y="215" text-anchor="middle" font-family="Arial, sans-serif" font-size="8" fill="#374151">Branch B</text>
  
  <circle cx="300" cy="310" r="15" fill="#f59e0b" stroke="#d97706" stroke-width="2"/>
  <text x="300" y="315" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" font-weight="bold" fill="white">3</text>
  <text x="300" y="335" text-anchor="middle" font-family="Arial, sans-serif" font-size="8" fill="#374151">Branch C</text>
  
  <!-- Node 4: Join Point -->
  <circle cx="420" cy="250" r="15" fill="#3b82f6" stroke="#2563eb" stroke-width="2"/>
  <text x="420" y="255" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" font-weight="bold" fill="white">4</text>
  <text x="420" y="280" text-anchor="middle" font-family="Arial, sans-serif" font-size="8" fill="#374151">Join</text>
  
  <!-- Node 5: Second Branch Point -->
  <circle cx="540" cy="250" r="15" fill="#3b82f6" stroke="#2563eb" stroke-width="2"/>
  <text x="540" y="255" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" font-weight="bold" fill="white">5</text>
  <text x="540" y="280" text-anchor="middle" font-family="Arial, sans-serif" font-size="8" fill="#374151">Gateway</text>
  
  <!-- SECOND BRANCH: (e|f) -->
  <circle cx="660" cy="190" r="15" fill="#ef4444" stroke="#dc2626" stroke-width="2"/>
  <text x="660" y="195" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" font-weight="bold" fill="white">6</text>
  <text x="660" y="215" text-anchor="middle" font-family="Arial, sans-serif" font-size="8" fill="#374151">Branch E</text>
  
  <!-- Node 7: Final Join -->
  <circle cx="780" cy="250" r="15" fill="#3b82f6" stroke="#2563eb" stroke-width="2"/>
  <text x="780" y="255" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" font-weight="bold" fill="white">7</text>
  <text x="780" y="280" text-anchor="middle" font-family="Arial, sans-serif" font-size="8" fill="#374151">Final</text>
  
  <!-- End Event -->
  <circle cx="880" cy="250" r="18" fill="#dc2626" stroke="#b91c1c" stroke-width="2"/>
  <text x="880" y="255" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" font-weight="bold" fill="white">E</text>
  <text x="880" y="280" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" fill="#374151">End</text>
  
  <!-- Sequential Flow Lines -->
  <line x1="98" y1="250" x2="162" y2="250" stroke="#374151" stroke-width="2" marker-end="url(#arrowhead)"/>
  <text x="130" y="245" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" fill="#374151">a</text>
  
  <!-- Branch Lines (b|c) -->
  <line x1="195" y1="240" x2="285" y2="200" stroke="#f59e0b" stroke-width="2" marker-end="url(#arrowhead)"/>
  <text x="235" y="215" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" fill="#f59e0b">b</text>
  
  <line x1="195" y1="260" x2="285" y2="300" stroke="#f59e0b" stroke-width="2" marker-end="url(#arrowhead)"/>
  <text x="235" y="285" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" fill="#f59e0b">c</text>
  
  <!-- Join Lines (c|b) - crossed pattern -->
  <line x1="315" y1="200" x2="405" y2="240" stroke="#374151" stroke-width="2" marker-end="url(#arrowhead)"/>
  <text x="355" y="215" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" fill="#374151">c</text>
  
  <line x1="315" y1="300" x2="405" y2="260" stroke="#374151" stroke-width="2" marker-end="url(#arrowhead)"/>
  <text x="355" y="285" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" fill="#374151">b</text>
  
  <!-- Sequential flow to second branch -->
  <line x1="435" y1="250" x2="522" y2="250" stroke="#374151" stroke-width="2" marker-end="url(#arrowhead)"/>
  <text x="478" y="245" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" fill="#374151">d</text>
  
  <!-- Second Branch Lines (e|f) -->
  <line x1="555" y1="240" x2="645" y2="200" stroke="#ef4444" stroke-width="2" marker-end="url(#arrowhead)"/>
  <text x="595" y="215" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" fill="#ef4444">e</text>
  
  <line x1="555" y1="260" x2="765" y2="250" stroke="#ef4444" stroke-width="2" marker-end="url(#arrowhead)"/>
  <text x="655" y="255" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" fill="#ef4444">f</text>
  
  <!-- Join Line from branch E -->
  <line x1="675" y1="200" x2="765" y2="240" stroke="#374151" stroke-width="2" marker-end="url(#arrowhead)"/>
  <text x="715" y="215" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" fill="#374151">f</text>
  
  <!-- Final sequential line to end -->
  <line x1="795" y1="250" x2="862" y2="250" stroke="#374151" stroke-width="2" marker-end="url(#arrowhead)"/>
  <text x="828" y="245" text-anchor="middle" font-family="Arial, sans-serif" font-size="10" fill="#374151">g</text>
  
  <!-- Arrow marker definition -->
  <defs>
    <marker id="arrowhead" markerWidth="10" markerHeight="7" refX="9" refY="3.5" orient="auto">
      <polygon points="0 0, 10 3.5, 0 7" fill="#374151"/>
    </marker>
  </defs>
  
  <!-- Legend positioned below the diagram -->
  <rect x="100" y="420" width="1400" height="220" fill="white" stroke="#e5e7eb" stroke-width="1" rx="5"/>
  <text x="800" y="445" text-anchor="middle" font-family="Arial, sans-serif" font-size="16" font-weight="bold" fill="#374151">SCF Process Components Legend</text>
  
  <!-- Legend items arranged horizontally in rows -->
  <!-- Row 1 -->
  <circle cx="150" cy="475" r="10" fill="#f59e0b"/>
  <text x="170" y="480" font-family="Arial, sans-serif" font-size="12" fill="#374151">First Branch (b|c): Two processing paths that cross-connect</text>
  
  <circle cx="700" cy="475" r="10" fill="#ef4444"/>
  <text x="720" y="480" font-family="Arial, sans-serif" font-size="12" fill="#374151">Second Branch (e|f): Alternative completion paths</text>
  
  <!-- Row 2 -->
  <circle cx="150" cy="505" r="10" fill="#3b82f6"/>
  <text x="170" y="510" font-family="Arial, sans-serif" font-size="12" fill="#374151">Sequential Steps: Process flow elements and join points</text>
  
  <circle cx="700" cy="505" r="10" fill="#10b981"/>
  <text x="720" y="510" font-family="Arial, sans-serif" font-size="12" fill="#374151">Start Event: Process initiation</text>
  
  <!-- Row 3 -->
  <circle cx="150" cy="535" r="10" fill="#dc2626"/>
  <text x="170" y="540" font-family="Arial, sans-serif" font-size="12" fill="#374151">End Event: Process completion</text>
  
  <circle cx="700" cy="535" r="10" fill="#6b7280"/>
  <text x="720" y="540" font-family="Arial, sans-serif" font-size="12" fill="#374151">Join Points: Branch convergence</text>
  
  <!-- Row 4 - Possible Paths -->
  <text x="150" y="570" font-family="Arial, sans-serif" font-size="12" font-weight="bold" fill="#374151">4 Possible Execution Paths:</text>
  <text x="150" y="590" font-family="Arial, sans-serif" font-size="11" fill="#374151">1. abcdefg (b→c→e→f) | 2. abcdfg (b→c→f) | 3. acbefg (c→b→e→f) | 4. acbfg (c→b→f)</text>
  
  <!-- Row 5 - Pattern Explanation -->
  <text x="150" y="615" font-family="Arial, sans-serif" font-size="12" font-weight="bold" fill="#374151">Pattern Logic: a(b|c)(c|b)d(e|f)fg</text>
  <text x="150" y="635" font-family="Arial, sans-serif" font-size="11" fill="#374151">Cross-connected branches ensure if first choice is 'b' then second is 'c', and vice versa</text>
</svg>