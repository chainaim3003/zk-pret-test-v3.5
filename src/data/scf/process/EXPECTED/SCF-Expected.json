{
  "processId": "SCF_Expected_Process",
  "processName": "Supply Chain Finance - Expected Pattern",
  "description": "Complete SCF workflow with cross-connected branching patterns",
  "processType": "SCF",
  "subType": "SUPPLY_CHAIN_FINANCE",
  "status": "EXPECTED",
  "version": "1.0",
  "expectedPattern": "a(b|c)(c|b)d(e|f)fg",
  "actualPath": "ALL_PATHS",
  "pathValidation": {
    "isValid": true,
    "matchesExpected": true,
    "complianceScore": 100,
    "possiblePaths": 4
  },
  "processFlow": {
    "startEvent": {
      "id": "Event_1e6mvux",
      "name": "0",
      "type": "startEvent"
    },
    "endEvent": {
      "id": "Event_0d8ci30", 
      "name": "8",
      "type": "endEvent"
    },
    "intermediateEvents": [
      {
        "id": "Event_0ckb2yy",
        "name": "1",
        "description": "Initial Processing Gateway",
        "type": "intermediateThrowEvent",
        "branches": ["Flow_1mcdovg", "Flow_0xad9v9"]
      },
      {
        "id": "Event_0orj063",
        "name": "2", 
        "description": "Branch B Processing",
        "type": "intermediateThrowEvent"
      },
      {
        "id": "Event_0s7vpsv",
        "name": "3",
        "description": "Branch C Processing",
        "type": "intermediateThrowEvent"
      },
      {
        "id": "Event_09jlpil",
        "name": "4",
        "description": "First Join Point",
        "type": "intermediateThrowEvent"
      },
      {
        "id": "Event_1ooatwj",
        "name": "5",
        "description": "Second Processing Gateway",
        "type": "intermediateThrowEvent",
        "branches": ["Flow_0u2xuqs", "Flow_0wslkb9"]
      },
      {
        "id": "Event_0zmkq57",
        "name": "6",
        "description": "Branch E Processing",
        "type": "intermediateThrowEvent"
      },
      {
        "id": "Event_024ls17",
        "name": "7",
        "description": "Final Join Point",
        "type": "intermediateThrowEvent"
      }
    ],
    "sequenceFlows": [
      {
        "id": "Flow_1e1jt09",
        "name": "a",
        "sourceRef": "Event_1e6mvux",
        "targetRef": "Event_0ckb2yy"
      },
      {
        "id": "Flow_1mcdovg",
        "name": "b",
        "sourceRef": "Event_0ckb2yy",
        "targetRef": "Event_0orj063"
      },
      {
        "id": "Flow_0xad9v9",
        "name": "c",
        "sourceRef": "Event_0ckb2yy",
        "targetRef": "Event_0s7vpsv"
      },
      {
        "id": "Flow_0uxr7s1",
        "name": "c",
        "sourceRef": "Event_0orj063",
        "targetRef": "Event_09jlpil"
      },
      {
        "id": "Flow_0q6gb9b",
        "name": "b",
        "sourceRef": "Event_0s7vpsv",
        "targetRef": "Event_09jlpil"
      },
      {
        "id": "Flow_16doo4t",
        "name": "d",
        "sourceRef": "Event_09jlpil",
        "targetRef": "Event_1ooatwj"
      },
      {
        "id": "Flow_0u2xuqs",
        "name": "e",
        "sourceRef": "Event_1ooatwj",
        "targetRef": "Event_0zmkq57"
      },
      {
        "id": "Flow_0wslkb9",
        "name": "f",
        "sourceRef": "Event_1ooatwj",
        "targetRef": "Event_024ls17"
      },
      {
        "id": "Flow_0564sbo",
        "name": "f",
        "sourceRef": "Event_0zmkq57",
        "targetRef": "Event_024ls17"
      },
      {
        "id": "Flow_01qx9za",
        "name": "g",
        "sourceRef": "Event_024ls17",
        "targetRef": "Event_0d8ci30"
      }
    ],
    "branches": [
      {
        "name": "First Branch",
        "pattern": "(b|c)",
        "description": "Initial routing choice",
        "options": [
          {
            "path": "b",
            "description": "Route through Event_0orj063 (Node 2)"
          },
          {
            "path": "c", 
            "description": "Route through Event_0s7vpsv (Node 3)"
          }
        ]
      },
      {
        "name": "Cross Connection",
        "pattern": "(c|b)",
        "description": "Cross-connected routing to join point",
        "logic": "If first choice was 'b', then use 'c'; if first choice was 'c', then use 'b'"
      },
      {
        "name": "Second Branch",
        "pattern": "(e|f)",
        "description": "Final routing choice",
        "options": [
          {
            "path": "e",
            "description": "Route through Event_0zmkq57 (Node 6) then 'f' to join"
          },
          {
            "path": "f",
            "description": "Direct route to Event_024ls17 (Node 7)"
          }
        ]
      }
    ],
    "possiblePaths": [
      {
        "pathId": 1,
        "sequence": "abcdefg",
        "description": "b→c cross-connection, then e→f processing"
      },
      {
        "pathId": 2,
        "sequence": "abcdfg",
        "description": "b→c cross-connection, then direct f"
      },
      {
        "pathId": 3,
        "sequence": "acbefg",
        "description": "c→b cross-connection, then e→f processing"
      },
      {
        "pathId": 4,
        "sequence": "acbfg",
        "description": "c→b cross-connection, then direct f"
      }
    ]
  },
  "businessRules": {
    "crossConnectionRequired": true,
    "branchingOptionsAvailable": true,
    "joinPointsRequired": true,
    "sequentialFlowMaintained": true,
    "complianceViolations": []
  },
  "metadata": {
    "createdDate": "2025-07-09",
    "lastModified": "2025-07-09",
    "version": "1.0",
    "author": "SCF Process Validator",
    "tags": ["scf", "supply-chain", "finance", "expected", "cross-connected", "branching"]
  }
}